--- origsrc/xapian-core-1.0.7/configure.ac	2008-07-15 23:42:35.000000000 -0500
+++ src/xapian-core-1.0.7/configure.ac	2008-07-16 20:42:35.984375000 -0500
@@ -127,6 +127,8 @@
 define([AC_LIBTOOL_LANG_RC_CONFIG], [:])
 
 dnl Use libtool to manage our libraries.
+AC_PROG_CC
+AC_PROG_CXX
 AC_PROG_LIBTOOL
 
 AM_CXXFLAGS=
@@ -216,8 +218,6 @@
   else
     CC_FOR_BUILD="$CC"
   fi
-  dnl Unset CC to make sure we don't accidentally use it.
-  unset CC
 fi
 AC_ARG_VAR(CC_FOR_BUILD, [C compiler command for native compilation (needed to compile build tools during cross-builds)])
 
